### ISSUE TITLE
As a library, setting the subscriber globally may not be very friendly to downstream users.

### ISSUE THREAD

**Issue by yw387659** [pvillela/rust-latency-trace#1]

**Comment by pvillela** [comment]
I agree this is not ideal. The alternative to setting a subscriber globally is to set it on each thread, using functions like [`tracing::dispatcher::set_default`](https://docs.rs/tracing/latest/tracing/dispatcher/fn.set_default.html) or [`tracing::dispatcher::with_default`](https://docs.rs/tracing/latest/tracing/dispatcher/fn.with_default.html). If the function `f` being instrumented spawns a thread, the code in the thread needs to execute in the scope of one of those `dispatcher` functions in order for the timing information for spans defined for `f`'s code in that thread to be collected. The `latency_trace` library has no way to introspect into `f` and inject calls to the aforementioned `dispatcher` functions. So, the only feasible solution I can see is to set the subscriber globally.

You will find that important examples like [`tracing_subscriber::fmt`](https://docs.rs/tracing-subscriber/latest/tracing_subscriber/fmt/index.html) (see [also](https://tokio.rs/tokio/topics/tracing)) and [`console_subscriber`](https://docs.rs/console-subscriber/latest/console_subscriber/) (see [also](https://tokio.rs/tokio/topics/tracing-next-steps)) set the subscriber globally as well.

I would be happy to consider a feasible alternative that does not require setting a global subscriber, if you can propose one.

**Comment by yw387659** [comment]
`Subscriber` can be a combination of `Layer`. As you gave the example, `tracing_subscriber::fmt` provides both [`Subscriber`](https://docs.rs/tracing-subscriber/latest/tracing_subscriber/fmt/fn.fmt.html) and [`Layer`](https://docs.rs/tracing-subscriber/latest/tracing_subscriber/fmt/fn.layer.html), and the latter can become part of the downstream combination Subscriber.
As a Library, whether it is providing `Subscriber` or `Layer`, I think `set_global_default`(`try_init`) should not be called

**Comment by pvillela** [comment]
Your point about `Layer`s is well taken. I have expanded the API by making `LatencyTrace` a `Layer` (the private type `LatencyTraceG`, which `LatencyTrace` wraps, was already a `Layer`) and adding two methods, `LatencyTrace::new` and `LatencyTrace::default` that can be used to instantiate a `LatencyTrace` without setting the global default subscriber. Downstream code can combine the `LatencyTrace` layer with other layers as needed. 

When the instrumented code is multi-threaded and cannot or should not be modified, `LatencyTrace` can be combined with other layers into a layered subscriber but someone will ultimately have to set that as the global default subscriber -- there is no way around it.

I have also added a couple of examples to show the combined use of `LatencyTrace` with `tracing_subscriber::fmt`. See the `develop` branch in the repo. I plan to update the documentation and push a new release with these changes in the next several days.
